PV260_Project/BusinessLayer/Services/Implementation/FileUrlService.cs

```diff
@@ -0,0 +1,47 @@
+﻿using DataLayer;
+using DataLayer.Models;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BusinessLayer.Services.Implementation
+{
+    public class FileUrlService : IFileUrlService
+    {
+        private readonly StockSystemDbContext _context;
+
+        public FileUrlService(StockSystemDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<FileUrl?> GetLatest()
+        {
+            return await _context.FileUrls.Where(fileUrl => fileUrl.ValidTo == null).FirstOrDefaultAsync();
+        }
+
+        public async Task<FileUrl> SetNewFileUrl(string url)
+        {
+            var transaction = await _context.Database.BeginTransactionAsync(isolationLevel: System.Data.IsolationLevel.Serializable);

```

Comment:
I have never seen this being used :D You think it is important? EF usually handles lots of stuff on its own.

Reply:
Handling transactions might not be important for such a simple app. I think it was also wrong so I changed it a bit, should I remove it?

Reply:
I haven't used these transaction even in more robust apps so I don't know the benefits of it being used like this. I would simply access the tables (DbSets) and work on them directly.